remotefs/download.go

```diff
@@ -0,0 +1,51 @@
+package remotefs
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+)
+
+// ErrChecksumMismatch is returned when the checksum of the uploaded file does not match the local checksum.
+
+// Download a file from the remote host.
+func Download(fs FS, src, dst string) error {
+	remote, err := fs.Open(src)
+	if err != nil {
+		return fmt.Errorf("open remote file for download: %w", err)
+	}
+	defer remote.Close()
+
+	remoteStat, err := remote.Stat()
+	if err != nil {
+		return fmt.Errorf("stat remote file for download: %w", err)
+	}
+
+	remoteSum := sha256.New()
+	localSum := sha256.New()
```

Comment:
This could check if the file exists locally and compare the shasums and just return nil if they match already 🤔